lib/Doctrine/ORM/Tools/Setup.php

```diff
@@ -145,7 +145,7 @@ public static function createConfiguration($isDevMode = false, $proxyDir = null,
             $cache = new ArrayCache();
         }
 
-        if ($cache instanceof CacheProvider) {
+        if ($cache instanceof CacheProvider && $cache->getNamespace() === '') {
```

Comment:
`=== ''` is not needed. Tests required tho

Reply:
I think that we should use the same approach of L2C where we always change change the namespace but using the current namespace as prefix, like:

https://github.com/doctrine/doctrine2/blob/b7cace86a078feb9d6da4f69fa4a49b99b353258/lib/Doctrine/ORM/Cache/DefaultCacheFactory.php#L234-L251

Reply:
not necessarily generating a new instance, though